### PR TITLE
feat(LayerManager): add a way to subscribe to stack changes

### PR DESCRIPTION
**Pull request by yndx-madfriend**
See https://github.com/gravity-ui/uikit/pull/492 for prequel

### CODE REVIEW

File: src/components/index.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ export * from './utils/withEventBrokerDomHandlers';
 export * from './utils/useBodyScrollLock';
 export * from './utils/useEventBroker';
 export * from './utils/useLayer';
+export * from './utils/LayerManager';
```
Comment (amje) [inline src/components/index.ts:59]: This file has exported private types. We don't want to expose them all

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/index.ts (hunk 1 above)
Comment (yndx-madfriend) [reply src/components/index.ts]: fixed

File: src/components/utils/LayerManager.ts [hunk 2]
```diff
@@ -39,6 +48,22 @@ class LayerManager {
...
+        return () => {
+            const index = this.stackChangeEventListeners.indexOf(listener);
+            if (index >= 0) {
+                this.stackChangeEventListeners.splice(index, 1);
+            }
+        };
+    }
+
+    getStackLength() {
+        return this.stack.length;
```
Comment (amje) [inline src/components/utils/LayerManager.ts:66]: Let's name it `getLayers` method

File: src/components/utils/LayerManager.ts [hunk 3]
```diff
@@ -53,6 +78,14 @@ class LayerManager {
         document.removeEventListener('mousedown', this.handleDocumentMouseDown, true);
     }
 
+    private notifyStackChanged() {
```
Comment (amje) [inline src/components/utils/LayerManager.ts:81]: `notifyLayersChange`

File: src/components/utils/LayerManager.ts [hunk 4]
```diff
@@ -53,6 +78,14 @@ class LayerManager {
...
+        for (const listener of this.stackChangeEventListeners) {
+            listener({
+                stackLength: this.getStackLength(),
```
Comment (amje) [inline src/components/utils/LayerManager.ts:84]: And here we can pass layers array as a single argument

**Review by amje — CHANGES_REQUESTED** [review]

**Review by ValeraS — APPROVED** [review]

**Review by amje — APPROVED** [review]

**Pull request merged**